internal/grpc/certificates.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+package grpc
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+)
+
+// appendRootCAs will read, parse, and append any certificates found in the
+// file at caFile to the RootCAs in the provided tls Config. If no filepath
+// is provided the tls Config is unmodified. By default, if there are no RootCAs
+// in the tls Config, it will automatically use the host OS's CA pool.
+func appendRootCAs(tlsConfig *tls.Config, caFile string) error {
+	if caFile == "" {
+		return nil
+	}
+
+	ca, err := os.ReadFile(caFile)
+	if err != nil {
+		return fmt.Errorf("could not read CA file (%s): %w", caFile, err)
```

Comment:
We need to change how we word our errors: https://github.com/uber-go/guide/blob/master/style.md#error-wrapping
```suggestion
		return fmt.Errorf("read CA file (%s): %w", caFile, err)
```
Ditto on line 35.